teuthology/task/kernel.py

```diff
@@ -1130,6 +1131,108 @@ def get_sha1_from_pkg_name(path):
     return sha1
 
 
+def process_role(ctx, config, timeout, role, role_config):
```

Comment:
I wonder if it make sense to leave the code of process after the task one so it is better seen what is changed.